### DISCUSSION TITLE
v8 dependency not listed in package.json

### DISCUSSION THREAD

**Discussion by HelpfulScripts** [panva/openid-client#442] [Q&A] [answered]
I am using openid-client 5.1.0 as a dependency of next-auth 4.x. When I start my app I get the following error message:
```
error - ./node_modules/openid-client/lib/helpers/deep_clone.js:1:33
Module not found: Can't resolve 'v8'

Import trace for requested module:
./node_modules/openid-client/lib/issuer.js
./node_modules/openid-client/lib/index.js
```

Sure enough, `deep_clone.js` uses a `require('v8')`, where 'v8' is not referenced in the `package.json` dependencies. I am wary to install `v8`, since all it seems to be needed for is the serialization/deserialization of an object. 

It seems that `next-auth` imports from `index.js`, which in turn appears to trigger the `require('v8')` call; Though`TokenSet` itself does not appear to depend on 'v8' in any way.
Is there a simple workaround to avoid the v8 dependency?
Many thanks!

**Comment by panva** [comment, marked as answer]
v8 is a standard node.js built-in module. It is most likely you're not using next-auth right, i'll defer to @balazsorban44

**Comment by HelpfulScripts** [comment]
...and I was using next-auth wrong. 
@panva: my bad, thanks for pointing me to the obvious :) 
Everything is fixed, works nicely.

**Reply by Batman4234** [reply]
what was wrong ?

**Comment by JTBrinkmann** [comment]
So obviously it's caused when trying to bundle server-side only code into a client bundle. I've had this issue when my client-side code imported a file with some helper functions, because that file also included an unrelated function which imported next-auth's `getServerSession` . Moving that unrelated function into a separate file fixed the issue.

**Comment by fullStackDataSolutions** [comment]
I updated to 
```
@auth0/nextjs-auth0": "^2.0.0"
```

And now get this error. There has to be some better way to let users know what to change to fix this if they upgrade.

**Reply by panva** [reply]
cc @adamjmcgrath

**Reply by JameelKhan9** [reply]
@blazestudios23 there were breaking changes in 2.0.0, client methods and components are now exported under /client:
https://github.com/auth0/nextjs-auth0/blob/v2.0.0/V2_MIGRATION_GUIDE.md

**Reply by adamjmcgrath** [reply]
Hi @blazestudios23 - yep as @JameelKhan9 has pointed out you need to import client methods/components from the /client export.

We had to split server and client parts of the SDK for Next.js 13 and React Server Side Components, so now when you import any of the client (browser) parts (as listed here https://github.com/auth0/nextjs-auth0#client-for-the-browser) you need to do the following:

```js
import {
  UserProvider,
  useUser,
  withPageAuthRequired // Note: there is also a server version of this
} from '@auth0/nextjs-auth0/client';
```
